src/main/java/com/ericsson/ei/frontend/utils/BackEndInstancesUtils.java

```diff
@@ -64,6 +66,53 @@
     private long nextTimeToParse = 0;
     private boolean savedSinceLastParsing = false;
 
+    /**
+     * Checks that all required keys exist in the incoming json object.
+     *
+     * @param JsonObject instance
+     * @return boolean
+     * */
+    public boolean hasRequiredJsonData(JsonObject instance) {
+        boolean hasNullValues = false;
+
+        // ensure required keys are present
+        if (instance.has(HOST) && instance.has(PORT) && instance.has(NAME)
+                && instance.has(CONTEXT_PATH) && instance.has(HTTPS)
+                && instance.has(DEFAULT)) {
+
+            // ensure JSON values are not null
+            for (Map.Entry<String, JsonElement> e : instance.entrySet()) {
+                if ((e.getValue().isJsonNull()) || e.getValue() == null) {
+                    hasNullValues = true;
+                }
+            }
+            // if no null values were found we return true
+            return (!hasNullValues) ? true : false;
+        }
+        return false;
+    }
+
+    /**
+     * Checks that the incoming json data does not contain additional keys. Any
+     * unrecognized keys are logged.
+     *
+     * @param JsonObject instance
+     * @return boolean
+     * */
+    public boolean containsAdditionalKeys(JsonObject instance) {
+        if (instance.size() > ALLOWED_JSON_KEYS.size()) {
+            for (Map.Entry<String, JsonElement> e : instance.entrySet()) {
+
+                // compare keys with valid keys
```

Comment:
Same here, if you feel like you should comment a if-statement. Create a variable in the line above with a good explainable name.

Reply:
Separated if statement into a variable.